src/STM32RTC.cpp

```diff
@@ -0,0 +1,814 @@
+  *      this list of conditions and the following disclaimer in the documentation
+  *      and/or other materials provided with the distribution.
+  *   3. Neither the name of STMicroelectronics nor the names of its contributors
+  *      may be used to endorse or promote products derived from this software
+  *      without specific prior written permission.
+  *
+  * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+  * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+  * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+  * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+  * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+  *
+  ******************************************************************************
+  */
+
+#include <time.h>
+
+#include "STM32RTC.h"
+
+#define EPOCH_TIME_OFF      946684800  // This is 1st January 2000, 00:00:00 in epoch time
+#define EPOCH_TIME_YEAR_OFF 100        // years since 1900
+
+// Initialize static variable
+bool STM32RTC::_configured = false;
+
+STM32RTC::STM32RTC(void)
+{
+
+}
+
+/**
+  * @brief initializes the RTC
+  * @param resetTime: if true reconfigures the RTC
+  * @param format: hour format: HOUR_12 or HOUR_24(default)
+  * @retval None
+  */
+void STM32RTC::begin(bool resetTime, hourFormat_t format)
+{
+  if(resetTime == true) {
+    _configured = false;
+  }
+  begin(format);
+}
+
+/**
+  * @brief initializes the RTC
+  * @param format: hour format: HOUR_12 or HOUR_24(default)
+  * @retval None
+  */
+void STM32RTC::begin(hourFormat_t format)
+{
+  if(_configured == false) {
+    RTC_init(format);
+  }
+
+  _hoursFormat = HOUR_AM;
+  _hours = 0;
+  _minutes = 0;
+  _seconds = 0;
+  _subSeconds = 0;
+  _year = 0;
+  _month = 0;
+  _date = 0;
+  _day = 0;
+  _alarmDate = 0;
+  _alarmHours = 0;
+  _alarmMinutes = 0;
+  _alarmSeconds = 0;
+  _alarmSubSeconds = 0;
+  _alarmFormat = HOUR_AM;
+
+  _configured = true;
+}
+
+/**
+  * @brief enable the RTC alarm.
+  * @param match: Alarm_Match configuration
+  * @retval None
+  */
+void STM32RTC::enableAlarm(Alarm_Match match)
+{
+  hourAM_PM_t format;
+  uint8_t date, hours, minutes, seconds;
+  uint32_t subSeconds;
+
+  if(_configured) {
+    RTC_GetTime(&hours, &minutes, &seconds, &subSeconds, &format);
+    date = getDay();
+
+    switch (match) {
+      case MATCH_OFF:
+        RTC_StopAlarm();
+        break;
+      case MATCH_YYMMDDHHMMSS://kept for compatibility
+      case MATCH_MMDDHHMMSS:  //kept for compatibility
```

Comment:
Where is the limitation about year / months described ?

Reply:
Inside the header file.